src/index.ts

```diff
@@ -84,22 +87,67 @@ const SchemaExtensionPlugin = makeExtendSchemaPlugin(build => {
           const {
             graphile: { fieldContext },
           } = resolveInfo;
-          return representations.map((representation: any) => {
+          return representations.map(async (representation: any) => {
             if (!representation || typeof representation !== "object") {
               throw new Error("Invalid representation");
             }
+
             const { __typename, [nodeIdFieldName]: nodeId } = representation;
-            if (!__typename || typeof nodeId !== "string") {
-              throw new Error("Failed to interpret representation");
+            if (!__typename) {
+              throw new Error(
+                "Failed to interpret representation, no typename"
+              );
+            }
+            if (nodeId) {
+              if (typeof nodeId !== "string") {
+                throw new Error(
+                  "Failed to interpret representation, invalid nodeId"
+                );
+              }
+              const x = resolveNode(
+                nodeId,
+                build,
+                fieldContext,
+                data,
+                context,
+                resolveInfo
+              );
+
+              return x;
+            } else {
+              const type = getTypeByName(__typename);
+              const { pgIntrospection: table } = scopeByType.get(type);
+
+              if (!table.primaryKeyConstraint) {
+                throw new Error("Failed to interpret representation");
+              }
+              const {
+                primaryKeyConstraint: { keyAttributes },
+              } = table;
+
+              const whereClause = sql.fragment`(${sql.join(
+                keyAttributes.map(
+                  (attr: PgAttribute) =>
+                    sql.fragment`${sql.identifier(attr.name)} = ${sql.value(
+                      representation[inflection.column(attr)]
+                    )}`
+                ),
+                ") and ("
+              )})`;
+
+              const rows = await resolveInfo.graphile.selectGraphQLResultFromTable(
```

Comment:
It's not possible to use `selectGraphQLResultFromTable` unless the field type is a table type/connection that was auto-generated by PostGraphile. In this case it's a new type `_Entity` so PostGraphile doesn't know how to plan that query.

Solving this is a little more complex; effectively you need to use the "row by unique constraint" plugin's resolver code:

https://github.com/graphile/graphile-engine/blob/1a8a0bc341ba21356ec57c4f19376c07a9ada6d4/packages/graphile-build-pg/src/plugins/PgRowByUniqueConstraint.js#L123-L153

Note the `TableType` you pass to `getDataFromParsedResolveInfoFragment` will change for each `getTypeByName` call you have here.

Reply:
That's the info I needed. Thanks!